contracts/rollup/Rollup.sol

```diff
@@ -546,31 +350,58 @@ contract Rollup is RollupHelpers {
             target.commitment.body.txs
         );
 
-        if (result != Types.Result.Ok) {
-            invalidBatchMarker = batchID;
-            slashAndRollback();
-        }
+        if (result != Types.Result.Ok) startRollingBack(batchID);
     }
+}
 
-    /**
-     * @notice Withdraw delay allows coordinators to withdraw their stake after the batch has been finalised
-     */
-    function withdrawStake(uint256 batchID) public {
-        require(
-            msg.sender == batches[batchID].committer(),
-            "You are not the correct committer for this batch"
+contract Rollup is RollupCore {
+    constructor(
```

Comment:
Move the constructor to the tail of the inheritance chain, so that testing is easier.